applications/ParticleMechanicsApplication/custom_elements/updated_lagrangian.cpp

```diff
@@ -1566,6 +1572,21 @@ void UpdatedLagrangian::CalculateOnIntegrationPoints(const Variable<double>& rVa
     else if (rVariable == MP_TOTAL_ENERGY) {
         rValues[0] = MPMEnergyCalculationUtility::CalculateTotalEnergy(*this);
     }
+    else if (rVariable == MP_HARDENING_RATIO) {
+        rValues[0] = mConstitutiveLawVector->GetValue(MP_HARDENING_RATIO, rValues[0]);
+    }
+    else if (rVariable == MP_EQUIVALENT_STRESS) {
+        rValues[0] = mConstitutiveLawVector->GetValue(MP_EQUIVALENT_STRESS, rValues[0]);
+    }
+    else if (rVariable == MP_EQUIVALENT_PLASTIC_STRAIN) {
+        rValues[0] = mConstitutiveLawVector->GetValue(MP_EQUIVALENT_PLASTIC_STRAIN, rValues[0]);
+    }
+    else if (rVariable == MP_EQUIVALENT_PLASTIC_STRAIN_RATE) {
+        rValues[0] = mConstitutiveLawVector->GetValue(MP_EQUIVALENT_PLASTIC_STRAIN_RATE, rValues[0]);
+    }
+    else if (rVariable == MP_TEMPERATURE) {
+        rValues[0] = mConstitutiveLawVector->GetValue(MP_TEMPERATURE, rValues[0]);
+    }
```

Comment:
You can put all those lines within a single one.

`
    else if (rVariable == MP_TEMPERATURE || ... ) {
        rValues[0] = mConstitutiveLawVector->GetValue(rVariable , rValues[0]);
    }
`

Reply:
good tip 👍